packages/flutter/test/gestures/drag_test.dart

```diff
@@ -528,14 +528,432 @@ void main() {
       'drag2-down',
       'drag2-cancel',
       '-d',
-      'drag1-update',
       '-e',
       'drag1-update',
       '-f',
       'drag1-end',
     ]);
   });
 
+  testGesture('A pointer joins and leave behavior', (GestureTester tester) {
+    final HorizontalDragGestureRecognizer drag1 =
+    HorizontalDragGestureRecognizer() ..dragStartBehavior = DragStartBehavior.down;
+    final VerticalDragGestureRecognizer drag2 =
+    VerticalDragGestureRecognizer() ..dragStartBehavior = DragStartBehavior.down;
+    addTearDown(() => drag1.dispose);
+    addTearDown(() => drag2.dispose);
+
+    Offset? updatedScrollDelta;
+    final List<String> log = <String>[];
+    drag1.onDown = (_) { log.add('drag1-down'); };
+    drag1.onStart = (_) { log.add('drag1-start'); };
+    drag1.onUpdate = (DragUpdateDetails details) {
+      updatedScrollDelta = details.delta;
+      log.add('drag1-update');
+    };
+    drag1.onEnd = (_) { log.add('drag1-end'); };
+    drag1.onCancel = () { log.add('drag1-cancel'); };
+    drag2.onDown = (_) { log.add('drag2-down'); };
+    drag2.onStart = (_) { log.add('drag2-start'); };
+    drag2.onUpdate = (_) { log.add('drag2-update'); };
+    drag2.onEnd = (_) { log.add('drag2-end'); };
+    drag2.onCancel = () { log.add('drag2-cancel'); };
+
+    // Now only one pointer moves.
+    final TestPointer pointer5 = TestPointer(5);
+    final PointerDownEvent down5 = pointer5.down(const Offset(10.0, 10.0));
+    drag1.addPointer(down5);
+    drag2.addPointer(down5);
+    tester.closeArena(5);
+    tester.route(down5);
+    expect(log, <String>[
+      'drag1-down',
+      'drag2-down',
+    ]);
```

Comment:
Can you clear the log after each `expect`?

Reply:
Done.